src/audio/detect_test.c

```diff
@@ -86,8 +86,8 @@ struct comp_data {
 	struct kpb_event_data event_data;
 	struct kpb_client client_data;
 
-	void (*detect_func)(struct comp_dev *dev,
-			    struct comp_buffer *source, uint32_t frames);
+	void (*detect_func)(const struct comp_dev *dev,
+			    const struct audio_stream *source, uint32_t frames);
 };
```

Comment:
Adding const should to structures should be a separate patch with a proper reasoning. We can even merge it earlier, then the rest of the patches if it makes sense.

Reply:
It is double work, and this changes do not firmware behavior - it's only compilation time restrictions.

Reply:
So maybe You merge https://github.com/thesofproject/sof/pull/2273 first, to this effort little smaller.

Reply:
@ktrzcinx done!

Reply:
@dbaluta thanks!